### PR TITLE
fix: Deprecation warning after running the development server

### LINKED ISSUE #1496: Deprecation warning after running the development server
**Describe the bug**
When the command ```npm run serve``` is executed to run the dev server, deprecation warnings are shown.

**To Reproduce**

1. Install the packages
2. Run ```npm run serve```

**Expected behavior**
There should be no warnings

**Actual behavior**
'onAfterSetupMiddleware' and 'onBeforeSetupMiddleware' are deprecated

**Screenshots**
image (copy 1)

**Additional details**

### PR DESCRIPTION
**Pull request by antriksh-9**
**What kind of change does this PR introduce?**
This PR fixes the deprecation warning of `onAfterSetupMiddleware` and `onBeforeSetupMiddleware`.

**Issue Number:**

Fixes #1496 

**Did you add tests for your changes?**
Not needed

**Snapshots/Videos:**

Screenshot 2024-03-12 183720

**If relevant, did you update the documentation?**
No

**Summary**

**Does this PR introduce a breaking change?**
No

**Other information**
I was told to replace `onAfterSetupMiddleware` and `onBeforeSetupMiddleware` function with `setupMiddlewares`.
But I didn't find these function in any of the file of `talawa-admin` repo. 
Most probably it should be present in `scripts/config-overrides/custom-start.js`.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@antriksh-9 Please fix the conflicting files.  Where are the rest of the checks?

**Comment by antriksh-9** [comment]
Hey @Cioppolo14 I have resolved the conflicts, but I have no idea where are the other checks.

**Comment by Cioppolo14** [comment]
@antriksh-9 They are there now.  I will assign reviewers.

**Comment by Cioppolo14** [comment]
@ttaylor92 @tasneemkoushar Can you review this PR?

**Comment by palisadoes** [comment]
We are going to pause merging PRs into the `develop` branch to facilitate fixing the roles of Members, Admins and Super Admins.

They need to assigned on a per organization basis versus per API instance. This is the tracking issue.

- https://github.com/PalisadoesFoundation/talawa-api/issues/1965

Reviews of PRs for all branches will continue

**Comment by palisadoes** [comment]
This is an update on the PR merging freeze:

1. In the next few hours we will be merging the `develop-userTypeFix` branch into the `develop` branch. 
1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. This merge will cause some conflicts in your PR.

We decided to do this at the beginning of the weekend to give us all time to adjust PR code and create bug fixes that may arise.

Update your code at or after midnight GMT on the morning of March 23, 2024. (5:30am IST). 

If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

**Comment by palisadoes** [comment]
## The PR merging freeze is lifted.

1. We are working on bug fixes that may arise. These should be few and are being tracked here:
    1. https://github.com/orgs/PalisadoesFoundation/projects/24
1. Please update your PRs and local repos. Fix any new merge conflicts that may have occurred.
    
Background:

1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

File: scripts/config-overrides/custom_start.js [hunk 1]
```diff
@@ -4,16 +4,69 @@ const { spawn } = require('child_process');
 const port = process.env.PORT || 4321;
 process.env.PORT = port;
 
-const react_script_start = 'npx react-scripts start';
-const react_app_rewired_start = 'npx react-app-rewired start --config-overrides=scripts/config-overrides';
+const reactAppRewiredStart =
+  'npx react-app-rewired start --config-overrides=scripts/config-overrides/';
 
-if (process.env.ALLOW_LOGS === "YES") {
-  // Execute the npm command
```
Comment (palisadoes) [inline scripts/config-overrides/custom_start.js:11]: Why was this logic removed?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comment

File: scripts/config-overrides/custom_start.js (hunk 1 above)
Comment (antriksh-9) [reply scripts/config-overrides/custom_start.js:11]: Hey @palisadoes I think it was removed by mistake, it should be present there.

**Comment by antriksh-9** [comment]
Hey @palisadoes I have made the changes.

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. I cannot log in using the sample data. This error is visible in the API console.
2. Please test your code more thoroughly

```
[0] error 2024-04-18 09:45:20 5q1tkapohi Cannot query field "userType" on type "User". {"locations":[{"line":5,"column":7}],"extensions":{"code":"GRAPHQL_VALIDATION_FAILED","stacktrace":["GraphQLError: Cannot query field \"userType\" on type \"User\".","    at Object.Field (/home/peter/code/GitHub/PalisadoesFoundation/talawa-api/node_modules/graphql/validation/rules/FieldsOnCorrectTypeRule.js:51:13)","    at Object.enter (/home/peter/code/GitHub/PalisadoesFoundation/talawa-api/node_modules/graphql/language/visitor.js:301:32)","    at Object.enter (/home/peter/code/GitHub/PalisadoesFoundation/talawa-api/node_modules/graphql/utilities/TypeInfo.js:391:27)","    at visit (/home/peter/code/GitHub/PalisadoesFoundation/talawa-api/node_modules/graphql/language/visitor.js:197:21)","    at validate (/home/peter/code/GitHub/PalisadoesFoundation/talawa-api/node_modules/graphql/validation/validate.js:91:24)","    at processGraphQLRequest (/home/peter/code/GitHub/PalisadoesFoundation/talawa-api/node_modules/@apollo/server/src/requestPipeline.ts:247:40)","    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","    at internalExecuteOperation (/home/peter/code/GitHub/PalisadoesFoundation/talawa-api/node_modules/@apollo/server/src/ApolloServer.ts:1313:12)","    at runHttpQuery (/home/peter/code/GitHub/PalisadoesFoundation/talawa-api/node_modules/@apollo/server/src/runHttpQuery.ts:232:27)","    at runPotentiallyBatchedHttpQuery (/home/peter/code/GitHub/PalisadoesFoundation/talawa-api/node_modules/@apollo/server/src/httpBatching.ts:85:12)","    at ApolloServer.executeHTTPGraphQLRequest (/home/peter/code/GitHub/PalisadoesFoundation/talawa-api/node_modules/@apollo/server/src/ApolloServer.ts:1087:14)"]}} 

```

File: scripts/config-overrides/custom_start.js (hunk 1 above)
Comment (palisadoes) [reply scripts/config-overrides/custom_start.js:11]: The logic is still removed. Why?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comment

**Comment by palisadoes** [comment]
Closing. Inactivity

**Pull request closed without merging**